src/js/recipe.js

```diff
@@ -0,0 +1,78 @@
+const ingredient = require("./ingredient");
+
+/**
+ * Creamos la clase receta.
+ */
+class Recipe {
+	/**
+	 * Recibe todos los parametros en el constructor.
+	 * Controlamos el tipo de los parametros que pasamos al constructor,
+	 * y si es de un tipo incorrecto al que queremos lanzamos una excepcion.
+	 * Si es correcto, se setea el parámetro.
+	 * @param idRecipe Integer
+	 * @param ingredients Array[Ingredients]
+	 * @param steps Array[Strings]
+	 * @param typeOfFood String
+	 * @param origin String
+	 * @param idAuthor Integer
+	 */
+	constructor(idRecipe, ingredients, steps, typeOfFood, origin, idAuthor) {
+		const tipoDatoInvalido = "El tipo de datos es inválido";
+		const datosInsuficinetes = "Datos insuficientes";
+		// Controlamos dato idRecipe
+		if (typeof idRecipe === typeof 1 && idRecipe > 0) {
+			this.idRecipe = idRecipe;
+		} else {
+			throw tipoDatoInvalido;
+		}
+		// Controlamos dato ingredients
+		if (ingredients instanceof Array) {
+			if (ingredients.length < 3) {
+				throw datosInsuficinetes;
+			}
+			for (let i = 0; i < ingredients.length; i += i) {
```

Comment:
Aquí está el fallo de por qué no termina el test: este bucle es infinito, debido al `i += i`. Cuando `i` es 0, se suma 0 a `i`, por lo tanto sigue siendo 0. Lo que hay que poner es `i += 1`, o `i++`.

Ya que estamos, te dejo un enlace a la [documentación de for...of](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Sentencias/for...of), que es una mejor forma de recorrer arrays cuando no necesitas saber los índices (como en este caso). Es como el `foreach` de PHP.